sktime/classification/early_classification/_teaser.py

```diff
@@ -0,0 +1,563 @@
+        new_state_info : 2D int array
+            An array containing the state info for each decision in X, contains
+            information for future decisions on the data. Returned if
+            return_safety_decisions is True.
+        """
+        self.check_is_fitted()
+
+        # boilerplate input checks for predict-like methods
+        X = self._check_convert_X_for_predict(X)
+
+        return self._predict_proba(X, state_info=state_info)
+
+    def _predict_proba(self, X, state_info=None):
+        n_instances, _, series_length = X.shape
+
+        # maybe use the largest index that is smaller than the series length
+        next_idx = self._get_next_idx(series_length)
+
+        if next_idx == -1:
+            raise ValueError(
+                f"Input series length does not match the classification points produced"
+                f" in fit. Current classification points: {self._classification_points}"
+            )
+
+        m = getattr(self.estimator, "n_jobs", None)
+        threads = self._threads_to_use if m is None else 1
+
+        # Always consider all previous time stamps up to the input series_length
+        if state_info is None or state_info == []:
+            state_info = np.zeros((n_instances, 3), dtype=int)
+        elif not all(si[0] == next_idx - 1 for si in state_info):
+            raise ValueError(
+                "All state_info input instances must be from the "
+                "previous classification point series length."
+            )
+
+        # determine last index used
+        last_idx = state_info[0, 0]
+
+        # compute all new updates since then
+        out = Parallel(n_jobs=threads)(
+            delayed(self._predict_proba_for_estimator)(
+                X,
+                i,
+            )
+            for i in range(last_idx, next_idx + 1)
+        )
+
+        X_oc, probas, preds = zip(*out)
+        new_state_info, accept_decision = self._predict_oc_classifier_n_timestamps(
+            preds, X_oc, self._consecutive_predictions, last_idx, next_idx + 1
+        )
+
+        probas = np.array(
+            [probas[new_state_info[i, 0] - last_idx][i] for i in range(n_instances)]
+        )
+
+        return (
+            (probas, accept_decision, new_state_info)
+            if self.return_safety_decisions
+            else probas
+        )
+
+    def _get_next_idx(self, series_length):
+        """Return the largest index smaller than the series length."""
+        next_idx = -1
+        for idx, offset in enumerate(np.sort(self._classification_points)):
+            if offset <= series_length:
+                next_idx = idx
+        return next_idx
+
+    def _fit_estimator(self, X, y, i):
+        rs = 255 if self.random_state == 0 else self.random_state
+        rs = None if self.random_state is None else rs * 37 * (i + 1)
+        rng = check_random_state(rs)
+
+        default = MUSE() if X.shape[1] > 1 else WEASEL()
```

Comment:
There is a problem with the default classifier:

> raise ValueError("Base estimator must have a predict_proba method.")
>      E           ValueError: Base estimator must have a predict_proba method.

Reply:
Never seen this before but changed the  _fit check, also accepts None.

Reply:
There is another potential problem with using the default classifier. Now we run into:

> E               ValueError: Found array with 0 feature(s) (shape=(20, 0)) while a minimum of 1 is required.

When running, i.e.:

```
teaser = TEASER(
        random_state=0,
        classification_points=[6, 10, 16, 24],
        return_safety_decisions=False,  # must be enforced?!
    )
```

I.e. if the classificaiton_point is smaller than the smallest weasel window size...

Reply:
I have set a min length of 8 if WEASEL/MUSE is used.

Reply:
Yes, noticed this issue at the same time :)